src/core_plugins/kibana/public/dashboard/top_nav/clone_modal.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  KuiModal,
+  KuiModalHeader,
+  KuiModalHeaderTitle,
+  KuiModalBody,
+  KuiModalBodyText,
+  KuiModalFooter,
+  KuiButton
+} from 'ui_framework/components';
+
+export class DashboardCloneModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newDashboardName: props.title
```

Comment:
Do you want the default value to cause a name collision with the existing dashboard title? Maybe `Clone` should be appended to the title or something like that, or maybe the default value is blank?